lib/phoenix/live_dashboard/helpers/supervision_tree_helpers/reingold_tifford_algorithm.ex

```diff
@@ -0,0 +1,150 @@
+defmodule Phoenix.LiveDashboard.ReingoldTilford do
+  # Reingold-Tilford algorithm for drawing trees
+  @moduledoc false
+
+  @height 20
+  @node_y_separation 10
+  @total_y_distance @height + @node_y_separation
+
+  def set_layout_settings(tree) do
+    tree
+    |> calculate_initial_y(0, [])
+    |> ensure_children_inside_screen()
+    |> put_final_y_values(0)
+  end
+
+  defp calculate_initial_y(%{children: children} = node, previous_sibling, top_siblings) do
+    {_, children} =
+      children
+      |> Enum.reduce({0, []}, fn n, {prev_sibling, nodes} ->
+        new_node = calculate_initial_y(n, prev_sibling, nodes)
+        {new_node.y, nodes ++ [new_node]}
+      end)
+
+    new_node =
+      case {node_type(node), node.num} do
+        {:leaf, 0} ->
+          %{node | y: 0}
+
+        {:leaf, _} ->
+          %{node | y: previous_sibling + @total_y_distance}
+
+        {:small_subtree, 0} ->
+          child = Enum.min_by(children, & &1.y)
+
+          %{node | y: child.y}
+
+        {:small_subtree, _} ->
+          %{
+            node
+            | y: previous_sibling + @total_y_distance,
+              modifier: previous_sibling + @total_y_distance + hd(children).y
+          }
+
+        {:big_subtree, 0} ->
+          mid = (List.last(children).y + hd(children).y) / 2
+          %{node | y: mid}
+
+        {:big_subtree, _} ->
+          mid = (List.last(children).y + hd(children).y) / 2
+
+          %{
+            node
+            | y: previous_sibling + @total_y_distance,
+              modifier: previous_sibling + @total_y_distance - mid
+          }
+      end
+
+    if !Enum.empty?(children) and !Enum.empty?(top_siblings) do
+      fix_sibling_conflicts(%{new_node | children: children}, top_siblings)
+    else
+      %{new_node | children: children}
+    end
+  end
+
+  defp node_type(node) do
+    with :subtree <- node.type,
+         1 <- Enum.count(node.children) do
```

Comment:
We don't need to traverse the whole list to get the whole count. Instead we can use pattern matching:

```suggestion
         [_] <- node.children do
```

Another suggestion though is to rewrite this to a cond:

```elixir
cond do
  node.type == :leaf -> :left
  match?([_], node.children) -> :small_subtree
  true -> :big_subtree
end
```